projects/fsastorefrontlib/src/core/product-pricing/facade/pricing.service.ts

```diff
@@ -25,4 +25,43 @@ export class PricingService {
     });
     return pricingAttributesData;
   }
+
+  buildPricingDataWithformDefinition(
```

Comment:
This method should be split into several separate methods for better overview and easier maintenance.